src/SuiTealiumTracker.js

```diff
@@ -0,0 +1,81 @@
+import { FunctionThrottler } from './FunctionThrottler'
+import { dispatchEvent } from '@s-ui/js/lib/events'
+
+const ELEMENTS_CLICKABLE = ['A', 'BUTTON', 'DIV', 'INPUT', 'SVG']
+const ELEMENT_NODE = 1
+const THROTTLE_WAIT = 100
+
+/**
+ * SuiTealiumTracker
+ */
+export class SuiTealiumTracker {
+  constructor ({customEventName, handleAnchorDelay} = {}) {
+    this.customEventName = customEventName
+    this.handleAnchorDelay = handleAnchorDelay
+    this.sendTealiumThrottled = FunctionThrottler.throttle(this.sendTealium, THROTTLE_WAIT)
+  }
+
+  init () {
+    this.initClickListener()
+    this.initChangeListener()
+    this.customEventName && this.promoteDispatchCustomEventToWindow()
+    this.customEventName && this.initCustomEventListener()
+  }
+
+  initClickListener () {
+    document.addEventListener('click', (e) => {
+      let target = e.target
+      let node = target
+      do {
+        if (node.nodeType !== ELEMENT_NODE) { break }
+        this.checkIfNeedToTrackClick({node, e})
+        const parentTagName = node.parentElement && node.parentElement.tagName.toUpperCase()
+        // search if the parent is a clickable element to check, otherwise we stop
+        node = ELEMENTS_CLICKABLE.includes(parentTagName) ? node.parentElement : false
+      } while (node)
+    })
+  }
+
+  initChangeListener () {
+    document.addEventListener('change', ({target}) => {
+      const {options = [], selectedIndex = 0} = target
+      const option = options[selectedIndex] || {dataset: {}}
+      const {tealiumTag} = option.dataset
+      if (tealiumTag) {
+        this.sendTealiumThrottled(tealiumTag)
+      }
+    })
+  }
+
+  initCustomEventListener () {
+    document.addEventListener(this.customEventName, (evt) => {
```

Comment:
I personally still miss a destroy function to remove all the listeners

Reply:
You don't need a destroy function as you want this listener in all your SPA. It's not like a component that has a listener that you might want to stop listening. The only important thing here is that `initCustomEventListener` should be called only once.

Reply:
As long as you don't made a require on a component-level context will be fine. The package is not changing the way that we currently are implementing the tracking service it is just putting it on a own repository. 

If you realise about if that is a real problem we could made a checker in the future.